<a href="https://colab.research.google.com/github/MuhammadFaisalAvicenna/datamining/blob/master/Notebook.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
%cd /content/drive/MyDrive/datamining/

/content/drive/MyDrive/datamining


In [1]:
!pip install jupyter-book

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     |████████████████████████████████| 43 kB 2.1 MB/s 
     |████████████████████████████████| 3.1 MB 64.3 MB/s 
     |████████████████████████████████| 56 kB 4.3 MB/s 
     |████████████████████████████████| 1.9 MB 51.5 MB/s 
     |████████████████████████████████| 41 kB 47 kB/s 
     |████████████████████████████████| 345 kB 80.8 MB/s 
     |████████████████████████████████| 46 kB 4.6 MB/s 
     |████████████████████████████████| 1.6 MB 62.3 MB/s 
     |████████████████████████████████| 70 kB 9.3 MB/s 
     |████████████████████████████████| 5.3 MB 48.7 MB/s 
     |████████████████████████████████| 41 kB 34 kB/s 
     |████████████████████████████████| 83 kB 2.4 MB/s 
     |████████████████████████████████| 60 kB 8.4 MB/s 
     |████████████████████████████████| 121 kB 64.5 MB/s 
     |████████████████████████████████| 84 kB 3.8 MB/s 
     |████████████████████████████████| 100 kB 9.8 

In [ ]:
!jupyter-book create tugas/

Traceback (most recent call last):
  File "/usr/local/bin/jupyter-book", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.7/dist-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.7/dist-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.7/dist-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.7/dist-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/jupyter_book/cli/main.py", line 340, in create
    sh.copytree(template_path, book)
  File "/usr/lib/python3.7/shutil.py", line 324, in copytree
    os.makedirs(dst)
  File "/usr/lib/python3.7/os.py", line 223,

In [ ]:
cd /content/drive/MyDrive/datamining

/content/drive/MyDrive/datamining


In [ ]:
!jupyter-book build tugas/

Running Jupyter-Book v0.13.1
Source Folder: /content/drive/MyDrive/datamining/tugas
Config Path: /content/drive/MyDrive/datamining/tugas/_config.yml
Output Path: /content/drive/MyDrive/datamining/tugas/_build/html
Running Sphinx v4.5.0
[etoc] Changing master_doc to 'intro'
loading pickled environment... checking bibtex cache... up to date
done
myst v0.15.2: MdParserConfig(renderer='sphinx', commonmark_only=False, enable_extensions=['colon_fence', 'dollarmath', 'linkify', 'substitution', 'tasklist'], dmath_allow_labels=True, dmath_allow_space=True, dmath_allow_digits=True, dmath_double_inline=False, update_mathjax=True, mathjax_classes='tex2jax_process|mathjax_process|math|output_area', disable_syntax=[], url_schemes=['mailto', 'http', 'https'], heading_anchors=None, heading_slug_func=None, html_meta=[], footnote_transition=True, substitutions=[], sub_delimiters=['{', '}'], words_per_minute=200)
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 0 so

In [ ]:
%cd /content/drive/MyDrive/datamining/Tugas

/content/drive/MyDrive/datamining/Tugas


In [ ]:
!git init

Reinitialized existing Git repository in /content/drive/MyDrive/datamining/Tugas/.git/


In [ ]:
get_token = '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'
repository = 'datamining'
username = 'hiisalvic'

In [ ]:
!git remote add origin https://hiisalvic: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@github.com/hiisalvic/datamining.git

fatal: remote origin already exists.


In [ ]:
!git config -l

core.repositoryformatversion=0
core.filemode=true
core.bare=false
core.logallrefupdates=true
remote.origin.url=https://hiisalvic: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@github.com/hiisalvic/datamining.git
remote.origin.fetch=+refs/heads/*:refs/remotes/origin/*


In [ ]:
!pip install ghp-import

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [ ]:
cd /content/drive/MyDrive/datamining/

/content/drive/MyDrive/datamining


In [ ]:
!jb build --all tugas/

Running Jupyter-Book v0.13.1
Source Folder: /content/drive/MyDrive/datamining/tugas
Config Path: /content/drive/MyDrive/datamining/tugas/_config.yml
Output Path: /content/drive/MyDrive/datamining/tugas/_build/html
Running Sphinx v4.5.0
[etoc] Changing master_doc to 'intro'
checking bibtex cache... out of date
parsing bibtex file /content/drive/MyDrive/datamining/tugas/references.bib... parsed 5 entries
myst v0.15.2: MdParserConfig(renderer='sphinx', commonmark_only=False, enable_extensions=['colon_fence', 'dollarmath', 'linkify', 'substitution', 'tasklist'], dmath_allow_labels=True, dmath_allow_space=True, dmath_allow_digits=True, dmath_double_inline=False, update_mathjax=True, mathjax_classes='tex2jax_process|mathjax_process|math|output_area', disable_syntax=[], url_schemes=['mailto', 'http', 'https'], heading_anchors=None, heading_slug_func=None, html_meta=[], footnote_transition=True, substitutions=[], sub_delimiters=['{', '}'], words_per_minute=200)
building [mo]: targets for 0 po 

In [ ]:
%cd /content/drive/MyDrive/datamining/tugas

/content/drive/MyDrive/datamining/tugas


In [ ]:
!ghp-import [-n] tugas/_build/html

usage: ghp-import [-h] [--version] [-n] [-c CNAME] [-m MESG] [-p] [-x PREFIX]
                  [-f] [-o] [-r REMOTE] [-b BRANCH] [-s] [-l]
                  directory
ghp-import: error: unrecognized arguments: tugas/_build/html


# Tugas 1

In [ ]:
import numpy as np
import pandas as pd

In [ ]:
dataset_url = "https://raw.githubusercontent.com/MuhammadFaisalAvicenna/datamining/master/train_and_test2.csv"
df = pd.read_csv(dataset_url, index_col='Passengerid')

In [ ]:
df

Age      Fare  Sex  sibsp  zero  zero.1  zero.2  zero.3  zero.4  \
Passengerid                                                                     
1            22.0    7.2500    0      1     0       0       0       0       0   
2            38.0   71.2833    1      1     0       0       0       0       0   
3            26.0    7.9250    1      0     0       0       0       0       0   
4            35.0   53.1000    1      1     0       0       0       0       0   
5            35.0    8.0500    0      0     0       0       0       0       0   
...           ...       ...  ...    ...   ...     ...     ...     ...     ...   
1305         28.0    8.0500    0      0     0       0       0       0       0   
1306         39.0  108.9000    1      0     0       0       0       0       0   
1307         38.5    7.2500    0      0     0       0       0       0       0   
1308         28.0    8.0500    0      0     0       0       0       0       0   
1309         28.0   22.3583    0      1     0       0       0       0       0   

             zero.5  ...  zero.12  zero.13  zero.14  Pclass  zero.15  zero.16  \
Passengerid          ...                                                        
1                 0  ...        0        0        0       3        0        0   
2                 0  ...        0        0        0       1        0        0   
3                 0  ...        0        0        0       3        0        0   
4                 0  ...        0        0        0       1        0        0   
5                 0  ...        0        0        0       3        0        0   
...             ...  ...      ...      ...      ...     ...      ...      ...   
1305              0  ...        0        0        0       3        0        0   
1306              0  ...        0        0        0       1        0        0   
1307              0  ...        0        0        0       3        0        0   
1308              0  ...        0        0        0       3        0        0   
1309              0  ...        0        0        0       3        0        0   

             Embarked  zero.17  zero.18  2urvived  
Passengerid                                        
1                 2.0        0        0         0  
2                 0.0        0        0         1  
3                 2.0        0        0         1  
4                 2.0        0        0         1  
5                 2.0        0        0         0  
...               ...      ...      ...       ...  
1305              2.0        0        0         0  
1306              0.0        0        0         0  
1307              2.0        0        0         0  
1308              2.0        0        0         0  
1309              0.0        0        0         0  

[1309 rows x 27 columns]

In [ ]:
number_of_columns = df.shape[1]

In [ ]:
pd.set_option('display.max_columns', number_of_columns)
pd.set_option('display.max_rows', number_of_columns)

In [ ]:
df.columns

Index(['Age', 'Fare', 'Sex', 'sibsp', 'zero', 'zero.1', 'zero.2', 'zero.3',
       'zero.4', 'zero.5', 'zero.6', 'Parch', 'zero.7', 'zero.8', 'zero.9',
       'zero.10', 'zero.11', 'zero.12', 'zero.13', 'zero.14', 'Pclass',
       'zero.15', 'zero.16', 'Embarked', 'zero.17', 'zero.18', '2urvived'],
      dtype='object')

#Nominal Features


*   Survived
*   Embarked



In [ ]:
df[["Age","2urvived", "Embarked"]].head(5)

Age  2urvived  Embarked
Passengerid                          
1            22.0         0       2.0
2            38.0         1       0.0
3            26.0         1       2.0
4            35.0         1       2.0
5            35.0         0       2.0

# Change Values 1/10


* Take all values of ‘Embarked’ series

* If values is 2.0 change to S

* If values is 0.0 change to C



In [ ]:
code_embark_for_cherbourg = "0"
code_embark_for_southampton = "2"

value_of_one = 1
value_of_zero = 0

def change_code_embark_to_biner(embarked):
    return value_of_one if embarked == code_embark_for_southampton else value_of_zero

In [ ]:
df["Embarked"] = df["Embarked"].apply(change_code_embark_to_biner)

In [ ]:
df[["Age","2urvived", "Embarked"]].head(5)

Age  2urvived  Embarked
Passengerid                          
1            22.0         0         0
2            38.0         1         0
3            26.0         1         0
4            35.0         1         0
5            35.0         0         0

In [ ]:
DECREMENT_BY_ONE = 1
INCREMENT_BY_ONE = 1

CONTINGENCY_TABLE_VALUE = {
    "q" : (1,1),
    "r" : (1,0),
    "s" : (0,1),
    "t" : (0,0),
}

In [ ]:
def get_series(df, idx, series):
    return df.loc[(idx), series]

In [ ]:
def get_dissimilarity_dataset(df, series_index = [], series = []):
    first_series = get_series(df, series_index[0], series)
    second_series = get_series(df, series_index[1], series)
    dataset = pd.concat([first_series,second_series],axis=1)
    return dataset.T

In [ ]:
get_dissimilarity_dataset(df, [1,2], ["2urvived", "Embarked"]).T

1    2
2urvived  0.0  1.0
Embarked  0.0  0.0

In [ ]:
df.loc[0:5, ["2urvived", "Embarked"]]

2urvived  Embarked
Passengerid                    
1                   0         0
2                   1         0
3                   1         0
4                   1         0
5                   0         0

In [ ]:
def count_contingency_value(df, start_index = 0, last_index = 1):

    CONTINGENCY_VALUE = {
        "q" : 0,
        "r" : 0,
        "s" : 0,
        "t" : 0,
    }

    column_range = df.shape[1]

    for column in range(column_range):
        for value in CONTINGENCY_TABLE_VALUE:
            item = list((tuple(df.loc[(start_index):(last_index), df.columns[column]]) == CONTINGENCY_TABLE_VALUE[value], value))
            if item[0] == True:
                if item[1] == "q":
                    CONTINGENCY_VALUE["q"] += 1
                if item[1] == "r":
                    CONTINGENCY_VALUE["r"] += 1
                if item[1] == "s":
                    CONTINGENCY_VALUE["s"] += 1
                if item[1] == "t":
                    CONTINGENCY_VALUE["t"] += 1

    return CONTINGENCY_VALUE

In [ ]:
df_1_2 = get_dissimilarity_dataset(df, [1,2], ["2urvived", "Embarked"])

In [ ]:
c_d_1_2 = count_contingency_value(df_1_2, 1, 2)

In [ ]:
df_1_3 = get_dissimilarity_dataset(df, [1,3], ["2urvived", "Embarked"])

In [ ]:
c_d_1_3 = count_contingency_value(df_1_3, 1, 3)

In [ ]:
df_1_4 = get_dissimilarity_dataset(df, [1,4], ["2urvived", "Embarked"])

In [ ]:
c_d_1_4 = count_contingency_value(df_1_4, 1, 4)

In [ ]:
df_1_5 = get_dissimilarity_dataset(df, [1,5], ["2urvived", "Embarked"])

In [ ]:
c_d_1_5 = count_contingency_value(df_1_5, 1, 5)

# Dissimilarity Binary Assymetric Value Formula


In [ ]:
def measure_dissimilarity_binary_value_assymetric_distance(contingency_value):

    return (contingency_value["r"] + contingency_value["s"]) / (contingency_value["q"] + contingency_value["r"] + contingency_value["s"])

In [ ]:
d_1_2 = measure_dissimilarity_binary_value_assymetric_distance(c_d_1_2)
d_1_3 = measure_dissimilarity_binary_value_assymetric_distance(c_d_1_3)
d_1_4 = measure_dissimilarity_binary_value_assymetric_distance(c_d_1_4)

In [ ]:
d_1_2

1.0

In [ ]:
d_1_3

1.0

In [ ]:
d_1_4

1.0

# Diskritisasi
# Load data Iris dataset
# Lakukan diskritisasi pada dataset

In [ ]:
import numpy as pd
import pandas as pd

In [ ]:
dataset_url='https://github.com/MuhammadFaisalAvicenna/datamining/blob/master/Iris.csv'
data=pd

In [ ]:
from sklearn import datasets
iris= datasets.load_iris()
x= iris.data[:, :1]
y= iris.target

In [ ]:
x

array([[5.1],
       [4.9],
       [4.7],
       [4.6],
       [5. ],
       [5.4],
       [4.6],
       [5. ],
       [4.4],
       [4.9],
       [5.4],
       [4.8],
       [4.8],
       [4.3],
       [5.8],
       [5.7],
       [5.4],
       [5.1],
       [5.7],
       [5.1],
       [5.4],
       [5.1],
       [4.6],
       [5.1],
       [4.8],
       [5. ],
       [5. ],
       [5.2],
       [5.2],
       [4.7],
       [4.8],
       [5.4],
       [5.2],
       [5.5],
       [4.9],
       [5. ],
       [5.5],
       [4.9],
       [4.4],
       [5.1],
       [5. ],
       [4.5],
       [4.4],
       [5. ],
       [5.1],
       [4.8],
       [5.1],
       [4.6],
       [5.3],
       [5. ],
       [7. ],
       [6.4],
       [6.9],
       [5.5],
       [6.5],
       [5.7],
       [6.3],
       [4.9],
       [6.6],
       [5.2],
       [5. ],
       [5.9],
       [6. ],
       [6.1],
       [5.6],
       [6.7],
       [5.6],
       [5.8],
       [6.2],
       [5.6],
       [5.9],
      

In [ ]:
df= pd.DataFrame(x,columns=['sepal'])

In [ ]:
df

sepal
0      5.1
1      4.9
2      4.7
3      4.6
4      5.0
..     ...
145    6.7
146    6.3
147    6.5
148    6.2
149    5.9

[150 rows x 1 columns]

# Tugas 3

## Baca Data 


*   Baca data dengan *library pandas* *italicized text*
*   Data [Bunga Iris](https://gist.githubusercontent.com/netj/8836201/raw/6f9306ad21398ea43cba4f7d537619d0e07d5ae3/iris.csv)


#### Persiapan :


*   Import *train_test_split* dari *sklearn*
*   Import *KNeighborsClassifier* dari *sklearn*
*   Import *pandas library*



In [ ]:
from sklearn.model_selection import train_test_split#split dataset into train and test data

In [ ]:
from sklearn.neighbors import KNeighborsClassifier# Create KNN classifier

In [ ]:
import pandas as pd

In [ ]:
dataset_url = "https://gist.githubusercontent.com/netj/8836201/raw/6f9306ad21398ea43cba4f7d537619d0e07d5ae3/iris.csv"

In [ ]:
df = pd.read_csv(dataset_url)
df.head()

sepal.length  sepal.width  petal.length  petal.width variety
0           5.1          3.5           1.4          0.2  Setosa
1           4.9          3.0           1.4          0.2  Setosa
2           4.7          3.2           1.3          0.2  Setosa
3           4.6          3.1           1.5          0.2  Setosa
4           5.0          3.6           1.4          0.2  Setosa

In [ ]:
#check number of rows and columns in dataset
df.shape

(150, 5)

### Hitung Data

*   Pisahkan kolom "variety" dari *dataframe*
*   Ambil kolom "variety" sebagai target kolom untuk kategori kelas
*   Pisahkan data latih dengan data tes


```
> data latih (nilai data)
X_train 

> data tes (nilai data)
X_test 

> data latih (kelas data)
y_train

> data tes (kelas data)
y_test
```





In [ ]:
#create a dataframe with all training data except the target column
X = df.drop(columns=["variety"])#check that the target variable has been removed
X.head()

sepal.length  sepal.width  petal.length  petal.width
0           5.1          3.5           1.4          0.2
1           4.9          3.0           1.4          0.2
2           4.7          3.2           1.3          0.2
3           4.6          3.1           1.5          0.2
4           5.0          3.6           1.4          0.2

In [ ]:
# separate target values
y = df["variety"].values  # view target values
y[0:5]

array(['Setosa', 'Setosa', 'Setosa', 'Setosa', 'Setosa'], dtype=object)

In [ ]:
# percent_amount_of_test_data = / HUNDRED_PERCENT
percent_amount_of_test_data = 0.2

In [ ]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = percent_amount_of_test_data, random_state=1, stratify=y)



*   Tentukan nilai K = 10 (10 tetangga terdekat)
*   Buat data latih


```
knn_classifier.fit(train_data)
```

In [ ]:
amount_of_neighbor = 3

In [ ]:
# Fit the classifier to the data
knn = KNeighborsClassifier(n_neighbors = amount_of_neighbor)

# create train data
knn.fit(X_train,y_train)

KNeighborsClassifier(n_neighbors=3)



*   Tes data target terhadap data latih
*   Tampilkan predksi 10 data pertama



```
knn_classifier.predict(test_data)
```





In [ ]:
#show first 5 model predictions on the test data
knn.predict(X_test)[0:30]

array(['Virginica', 'Setosa', 'Versicolor', 'Setosa', 'Setosa', 'Setosa',
       'Virginica', 'Virginica', 'Virginica', 'Versicolor', 'Setosa',
       'Versicolor', 'Virginica', 'Versicolor', 'Virginica', 'Setosa',
       'Virginica', 'Versicolor', 'Versicolor', 'Virginica', 'Versicolor',
       'Versicolor', 'Setosa', 'Setosa', 'Virginica', 'Versicolor',
       'Setosa', 'Setosa', 'Versicolor', 'Versicolor'], dtype=object)



*   Cetak prediksi nilai dari data tes



```
knn_classifier.score(test_data)
```





In [ ]:
#check accuracy of our model on the test data
knn.score(X_test, y_test)

0.9666666666666667

### Kesimpulan


*   Dataset bunga *iris* terdiri dari 150 data
*   Ambil 80% dari *iris dataset* sebagai data latih
*   Ambil 20% dari *iris dataset* sebagai data tes
*   Tetapkan nilai K = 10
*   Hasil 10 data pertama dari 20% data latih


```
['Virginica', 'Setosa', 'Versicolor', 'Setosa', 'Setosa', 'Setosa',
       'Virginica', 'Virginica', 'Virginica', 'Versicolor']
```
*   Dengan akurasi data yaitu $ 0.967\% $



